### PR TITLE
WIP – Print more measurements of student profile timing

### PR DESCRIPTION
**Pull request by alexsoble**
# Notes 

+ @kevinrobinson, this branch is my first WIP crack at investigating #1386. My plan is to find the student who triggered Logentry alerts, copy their data down locally, and use this profiling to see which parts of assembling `@serialized_data` take the most time on the server. 
+ What do you think about eventually shipping some version of this branch to production? Then we'd have lots of fine-grained student profile timing data in the logs on an ongoing basis.

### CODE REVIEW

**Comment by kevinrobinson** [comment]
@alexsoble Yeah, that seems like a good idea!  I think the most important thing that came up from the data in https://github.com/studentinsights/studentinsights/pull/1400#issue-290304976 is the variability.  Some hypotheses about why that might be are that it's from:

1. particular sets of data taking much longer than others (eg, a particular student has lots of data)
2. from changes because of increased load at different times
3. from just inherent variability in our Heroku setup

I added another commit on https://github.com/studentinsights/studentinsights/pull/1400#issuecomment-359641676 to see if there were particular students that were consistently slower.  Looking at individual paths, nothing jumped out at me (eg, paths that were slow at times were fast at others).  But graphing the time vs. latency looked like perhaps the slower requests are happening at similar times (eg, maybe because of concurrent requests).  I also tried running `ab` with a few requests to see if we had large variability at one set of time.  I didn't see anything surprising there except for when there are concurrent requests.

So I think looking at how to make these requests do less work is a great idea.  We could split these into separate requests (eg, don't load STAR math data for charts until the user navigates there, or don't load the "full case history" until they navigate there).  Those might be worth trying as well.

File: app/controllers/students_controller.rb [hunk 1]
```diff
@@ -24,27 +24,70 @@ def authorize_for_districtwide_access_admin
 
   def show
     student = Student.find(params[:id])
-    chart_data = StudentProfileChart.new(student).chart_data
+
+    serialized_student = log_timing('students#serialize_student_for_profile') do
```
Comment (kevinrobinson) [inline app/controllers/students_controller.rb]: How about `students#show serialize_student_for_profile`?  That way it's clear it's a step within the `students#show` controller/action.

Comment (alexsoble) [reply app/controllers/students_controller.rb]: 👍

**Comment by alexsoble** [comment]
@kevinrobinson Ship this or close it? Slowness in the student profile page doesn't seem to be a big issue anymore and we haven't heard slowness complaints since we shipped the memory improvements, so I could go either way.

**Comment by kevinrobinson** [comment]
It seems great to get more latency data in general.  But I think it depends on whether we think we'll take action on that data in the next week or two.  If so, let's ship it and do that!  And if not, it's easy to add in this great kind of profiling when we do want to do something about it, so nothing much is lost if we close.

**Comment by alexsoble** [comment]
@kevinrobinson OK! I don't think we'll take action in the next week or two so I'll close.

**Pull request closed without merging**